src/main/java/org/example/PhoneBook.java

```diff
@@ -0,0 +1,174 @@
+        } else {
+            editOrganization(scanner, (OrganizationContact) contact);
+        }
+    }
+
+    static void editPerson(Scanner scanner, PersonContact contact) {
+        System.out.println(EDIT_PERSON_FIELD);
+        String input = scanner.nextLine();
+        switch (input) {
+            case "name" -> {
+                System.out.println("Enter name:");
+                String newName = scanner.nextLine();
+                contact.setName(newName);
+                System.out.println("The record updated!");
+                contact.setLastEditDate(LocalDateTime.now());
+            }
+            case "surname" -> {
+                System.out.println("Enter surname:");
+                String newSurname = scanner.next();
+                contact.setSurname(newSurname);
+                System.out.println("The record updated!");
+                contact.setLastEditDate(LocalDateTime.now());
+            }
+            case "gender" -> {
+                System.out.println("Enter gender:");
+                String newGender = scanner.next();
+                contact.setGender(newGender);
+                System.out.println("The record updated!");
+                contact.setLastEditDate(LocalDateTime.now());
+            }
+            case "number" -> {
+                scanner.nextLine();
+                System.out.println("Enter number:");
+                String newNumber = scanner.nextLine();
+                contact.setPhoneNumber(newNumber);
+                System.out.println("The record updated!");
+                contact.setLastEditDate(LocalDateTime.now());
+            }
+        }
+    }
+
+    static void editOrganization(Scanner scanner, OrganizationContact contact) {
+        System.out.println(EDIT_ORGANIZATION_FIELD);
+        String input = scanner.next();
+        switch (input) {
+            case "address" -> {
+                System.out.println("Enter address:");
+                scanner.nextLine();
+                String newAddress = scanner.nextLine();
+                contact.setAddress(newAddress);
+                System.out.println("The record updated!");
+            }
+            case "number" -> {
+                scanner.nextLine();
+                System.out.println("Enter number:");
+                String newNumber = scanner.nextLine();
+                contact.setPhoneNumber(newNumber);
+                System.out.println("The record updated!");
+            }
+        }
+    }
+
+    static void getInfo(int index) {
+        Contact contact = contactList.get(index);
+        System.out.println(contact.print());
+    }
+
+    public static boolean checkGender(String gender) {
+        return gender.equals("M") || gender.equals("F");
+    }
+
+    public static List<Contact> searchQuery(String substring) {
+        List<Contact> result = new ArrayList<>();
+        for (Contact contact : contactList) {
+            if (contact instanceof PersonContact) {
```

Comment:
hmm, isn't these if/else_if same? also the whole idea of OOP in this case could be not to use instanceof. can't we somehow refactor this functionality with polymorphism for instance?

Reply:
I didn't use polymorphism here, as the fields are different, for example to be able to call getGender() I have to be sure that, it's person who has gender